### PR TITLE
Send NOI Submission and Cancelled Emails

### PR DESCRIPTION
**Pull request by sandratoh**
- Create ParentType enum
- Add NOI templates
- Resolve circular imports errors in NOI-related modules with `forwardRef`
- Update email service with NOI methods and type
- Rename application related methods in email service

### CODE REVIEW

File: services/apps/alcs/src/providers/email/email.module.ts [hunk 1]
```diff
@@ -5,13 +5,17 @@ import { EmailStatus } from './email-status.entity';
...
+import { NoticeOfIntentModule } from '../../alcs/notice-of-intent/notice-of-intent.module';
+import { NoticeOfIntentSubmissionModule } from '../../portal/notice-of-intent-submission/notice-of-intent-submission.module';
 
 @Module({
   imports: [
     HttpModule,
     TypeOrmModule.forFeature([EmailStatus]),
     forwardRef(() => ApplicationModule),
     forwardRef(() => ApplicationSubmissionModule),
+    forwardRef(() => NoticeOfIntentModule),
```
Comment (dhaselhan) [inline services/apps/alcs/src/providers/email/email.module.ts:17]: Is this the part that started the forwardRef cascade?

Comment (sandratoh) [reply services/apps/alcs/src/providers/email/email.module.ts:17]: Yes it was...

File: services/apps/alcs/src/common/dtos/base.dto.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
 import { AutoMap } from '@automapper/classes';
 import { IsString, MaxLength, MinLength } from 'class-validator';
 
+export enum ParentType {
```
[the remark this answers is not in the document]
Comment (sandratoh) [reply services/apps/alcs/src/common/dtos/base.dto.ts:4]: This is also being used in card-subtask.dto, is it okay to be in email-status?

[the remark this answers is not in the document]
Comment (dhaselhan) [reply services/apps/alcs/src/common/dtos/base.dto.ts:4]: ahh good point, in that case maybe move it to card-subtask.dto?

[the remark this answers is not in the document]
Comment (sandratoh) [reply services/apps/alcs/src/common/dtos/base.dto.ts]: Updated!

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**